backend/routers/social_media_marketing_router.py

```
@@ -870,3 +870,179 @@ async def get_campaigns_data(status: Optional[str], platform: Optional[str]) ->
+        
+    except Exception as e:
+        logger.error(f"❌ Content generation failed: {e}")
+        return StandardResponse(
+            success=False,
+            message=f"Content generation failed: {str(e)}",
+            data={"error": str(e)}
+        ).dict()
+
+@social_marketing_router.post("/execute-posting")
+async def execute_posting(
+    admin_user: dict = Depends(get_admin_user)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)